src/Microsoft.DotNet.ProjectModel/Files/IncludeEntry.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.DotNet.ProjectModel.Files
+{
+    public class IncludeEntry : IEquatable<IncludeEntry>
+    {
+        public string TargetPath { get; }
+
+        public string SourcePath { get; }
+
+        public bool IsCustomTarget { get; set; }
+
+        public IncludeEntry(string target, string source)
+        {
+            TargetPath = target;
+            SourcePath = source;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals((IncludeEntry)obj);
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
```

Comment:
Looks like the `IncludeEntry` is used in `HashSet` https://github.com/dotnet/cli/pull/2493/files#diff-d3a7e6bddbc6e6c7b1bd24ad2c893bb4R29.

So better generate a hash code based on it's properties.